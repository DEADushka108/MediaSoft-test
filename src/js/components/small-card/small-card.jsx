import React, {useState} from 'react';
import PropTypes from 'prop-types';
import {redirectToRoute} from '../../store/redirect/redirect-action';
import {connect} from 'react-redux';
import {AppRoute} from '../../utils/const';
import {Link} from 'react-router-dom';
import {ActionCreator} from '../../store/ships/ships';
import {shipDetails} from '../../types/ships';

const SmallCard = (props) => {
  const {ship, redirect, onAddCartClick} = props;
  const {id, name, previewImage, price, isInCart} = ship;
  const [favorite, setFavorite] = useState(isInCart);

  return <article className="small-card">
    <div className="small-card__image" onClick={() => {
      redirect(`${AppRoute.CARD}/${id}`);
    }}>
      <img src={previewImage} className="small-card__img"/>
    </div>
    <div className="small-card__wrapper">
      <h3 className="small-card__name">
        <Link to={`${AppRoute.CARD}/${id}`} className="small-card__link">{name}</Link>
      </h3>
      <p className="small-card__price">{price}$</p>
    </div>
    <button className="small-card__button" type="button" onClick={() => {
      onAddCartClick(ship);
      setFavorite(!favorite);
    }}>{favorite ? `In cart` : `Add`} </button>
  </article>;
};

SmallCard.propTypes = {
  ship: shipDetails,
  redirect: PropTypes.func.isRequired,
  onAddCartClick: PropTypes.func.isRequired,
};

const mapDispatchToProps = (dispatch) => ({
  redirect(route) {
    dispatch(redirectToRoute(route));
  },
  onAddCartClick(ship) {
    ship.isInCart = !ship.isInCart;
    dispatch(ActionCreator.updateShipStatus(ship));
  },
});

export {SmallCard};
export default connect(null, mapDispatchToProps)(SmallCard);
