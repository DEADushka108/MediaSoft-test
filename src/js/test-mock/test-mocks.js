export * from './const/const';
export * from './ships/ships';
export * from './store/store';
