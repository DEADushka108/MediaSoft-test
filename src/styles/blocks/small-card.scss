.small-card {
  display: flex;
  justify-content: space-around;
  flex-direction: column;
  width: 320px;
  min-height: 500px;
  margin-top: 20px;
  margin-bottom: 20px;
  box-shadow: inset 1px rgba(13, 46, 68, 0.2);

  &:hover {
    border: none;
    box-shadow: 5px 20px 50px rgba(13, 46, 68, 0.2);
  }
}

.small-card__wrapper {
  width: 280px;
  align-self: center;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
}

.small-card__image {
  margin-top: 20px;
  align-self: center;
}

.small-card__link {
  text-decoration: none;
}

.small-card__button {
  margin-bottom: 20px;
  align-self: center;
  width: 281px;
  text-decoration: none;
  display: inline-block;
  vertical-align: middle;
  box-sizing: border-box;
  padding: 11px 12px 14px 12px;
  border-radius: 10px;
  text-align: center;
  color: $text-white;
  font-weight: 700;
  background-color: $button-blue;
  border: none;
  cursor: pointer;
  box-shadow: 5px 10px 20px rgba(53, 110, 173, 0.2);

  &:hover,
  &:focus {
    color: $text-blue;
    background-color: transparent;
    padding: 8px 9px 11px 9px;
    border: 3px solid $text-blue;
  }

  &:active {
    color: rgba($text-blue, 0.3);
  }
}

@media (min-width: $tablet-width) {
  .small-card__link {
    font-size: 20px;
  }

  .small-card__price {
    font-size: 20px;
  }

  .small-card__wrapper {
    align-items: center;
  }
}

@media (min-width: $desktop-width) {
  .small-card {
    margin: 20px;
  }
}