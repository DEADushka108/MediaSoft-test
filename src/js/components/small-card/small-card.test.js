import React from 'react';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import {store, ship, noop} from '../../test-mock/test-mocks';
import {MemoryRouter} from 'react-router-dom';
import {SmallCard} from './small-card';

it(`Should SmallCard component render correctly`, () => {
  const tree = renderer
    .create((
      <Provider store={store}>
        <MemoryRouter>
          <SmallCard
            ship={ship}
            redirect={noop}
            onAddCartClick={noop}
          />
        </MemoryRouter>
      </Provider>
    ), {
      createNodeMock() {
        return {};
      }
    })
    .toJSON();

  expect(tree).toMatchSnapshot();
});
