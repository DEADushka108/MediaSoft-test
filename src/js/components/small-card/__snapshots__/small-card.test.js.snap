// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should SmallCard component render correctly 1`] = `
<article
  className="small-card"
>
  <div
    className="small-card__image"
    onClick={[Function]}
  >
    <img
      className="small-card__img"
      src="./img/content/battlestar-galactica-preview.png"
    />
  </div>
  <div
    className="small-card__wrapper"
  >
    <h3
      className="small-card__name"
    >
      <a
        className="small-card__link"
        href="/card/3"
        onClick={[Function]}
      >
        Battlestar Galactica
      </a>
    </h3>
    <p
      className="small-card__price"
    >
      234122
      $
    </p>
  </div>
  <button
    className="small-card__button"
    onClick={[Function]}
    type="button"
  >
    Add
     
  </button>
</article>
`;
